src/hssm/param/regression_param.py

```diff
@@ -0,0 +1,402 @@
+"""A class for regression parameterization of HSSM models."""
+
+import logging
+import re
+from typing import Any, Literal, cast
+
+import bambi as bmb
+import numpy as np
+import pandas as pd
+from formulae import design_matrices
+
+from ..link import Link
+from ..prior import get_default_prior, get_hddm_default_prior
+from .param import Param
+from .user_param import UserParam
+
+_logger = logging.getLogger("hssm")
+
+
+class RegressionParam(Param):
+    """A regression parameter.
+
+    A regression parameter is a parameter that has a formula.
+
+    Parameters
+    ----------
+    name
+        The name of the parameter.
+    prior
+        The prior specification for the parameter.
+    formula
+        The formula for the parameter.
+    link
+        The link function for the parameter.
+    bounds
+        The bounds for the parameter.
+    user_param
+        The parameter as specified by the user.
+
+    Attributes
+    ----------
+    name
+    prior
+    formula
+    link
+    bounds
+    user_param
+    is_regression
+        Whether the parameter is a regression.
+    is_fixed
+        Whether the parameter is fixed.
+    is_vector
+        Whether the parameter is a vector parameter.
+    is_parent
+        Whether the parameter is a parent parameter.
+
+    Methods
+    -------
+    from_defaults
+        Create a RegressionParam object with default values.
+    fill_defaults
+        Fill in the default values for the parameter.
+    validate
+        Validate the parameter.
+    process_prior
+        Process the prior specification.
+    set_loglogit_link
+        Override the default link function.
+    make_safe_priors
+        Override the default priors.
+    reformat_formula
+        Reformat the formula.
+    """
+
+    terms: list[str]
```

Comment:
I'm a bit confused by this. Is it just a type hint for `terms` for a type checker?

Reply:
Yes, for some reason mypy cannot figure out types of properties in the parent class, so we are being specific here